.github/workflows/testing.yml

```diff
@@ -26,26 +26,63 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
 
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15'
+
+      - name: Setup Kube in Docker
+        uses: engineerd/setup-kind@v0.4.0
+
+      - name: Setup latest Kangal release
+        run: |
+          mkdir kangal && cd kangal
+          VERSION=$(curl -fsSL "https://api.github.com/repos/hellofresh/kangal/releases" | jq --raw-output 'map(select(.tag_name|startswith("kangal")|not))[0] | .tag_name')
+          KANGAL_SOURCE_URL="https://github.com/hellofresh/kangal/archive/${VERSION}.tar.gz"
+          curl -X GET -H "Accept:application/octet-stream" -fsSL "$KANGAL_SOURCE_URL" > "/tmp/kangal-source.tar.gz"
+          tar -xzf "/tmp/kangal-source.tar.gz" -C /tmp
+          mv "/tmp/kangal-${VERSION}"/* .
+          make apply-crd
+          sed -i "s/latest/local/" pkg/backends/jmeter/jmeter.go
+          sed -i "s/Always/IfNotPresent/" pkg/backends/jmeter/resources.go
+          make build
+
       - name: Build Docker base image
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v1.1.0
         with:
-          tags: hellofreshtech/kangal-jmeter:latest
-          context: docker/jmeter-base
-          file: docker/jmeter-base/Dockerfile
+          repository: hellofreshtech/kangal-jmeter
+          tags: local
+          path: docker/jmeter-base
           push: false
 
       - name: Build Docker master image
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v1.1.0
         with:
-          tags: hellofreshtech/kangal-jmeter-master:latest
-          context: docker/jmeter-master
-          file: docker/jmeter-master/Dockerfile
+          repository: hellofreshtech/kangal-jmeter-master
+          tags: local
+          path: docker/jmeter-master
           push: false
+          build_args: VERSION=local
 
       - name: Build Docker worker image
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v1.1.0
         with:
-          tags: hellofreshtech/kangal-jmeter-worker:latest
-          context: docker/jmeter-worker
-          file: docker/jmeter-worker/Dockerfile
+          repository: hellofreshtech/kangal-jmeter-worker
+          tags: local
+          path: docker/jmeter-worker
           push: false
+          build_args: VERSION=local
+
+      - name: Load docker images into kind cluster
+        run: |
+          docker images
+          kind load docker-image hellofreshtech/kangal-jmeter-master:local
+          kind load docker-image hellofreshtech/kangal-jmeter-worker:local
+
+      - name: Run Integration Test
+        env:
+          AWS_ENDPOINT_URL: "localhost:8081"
+          AWS_BUCKET_NAME: "kangal-test"
+        run: |
+          cd kangal
+          ./ci/integration-tests.sh
```

Comment:
That means if we change Kangal structure it breaks the current master in this repo.
I am a bit worried about this approach because looks like we are running same integration tests from two places: one in Kangal, second is here.

Reply:
> That means if we change Kangal structure it breaks the current master in this repo.

Because we have the same test in `kangal`, it would fail and get caught before it gets merged.

>  I am a bit worried about this approach because looks like we are running same integration tests from two places: one in Kangal, second is here.

This is intentional; I don't think it's something to be worried about 👀